src/Sylius/Bundle/ApiBundle/CommandHandler/PickupCartHandler.php

```diff
@@ -85,11 +85,22 @@ public function __invoke(PickupCart $pickupCart)
 
         /** @var LocaleInterface $locale */
         $locale = $channel->getDefaultLocale();
+
+        /** @var string $localeCode */
+        $localeCode = $locale->getCode();
+
+        /** @var string|null $commandLocaleCode */
+        $commandLocaleCode = $pickupCart->localeCode;
+
+        if ($commandLocaleCode !== null && $channel->hasLocaleWithLocaleCode($commandLocaleCode)) {
+            $localeCode = $commandLocaleCode;
+        }
```

Comment:
```suggestion
        /** @var string|null $commandLocaleCode */
        $localeCode = $pickupCart->localeCode;

        if ($$localeCode !== null && !$channel->hasLocaleWithLocaleCode($localeCode)) {
            /** @var LocaleInterface $locale */
            $locale = $channel->getDefaultLocale();
            
            $localeCode = $locale->getCode();
        }
```

This should be also documented on API endpoint